### DISCUSSION TITLE
NativePHP and Laravel Queue

### DISCUSSION THREAD

**Discussion by mongobird** [NativePHP/laravel#241] [Q&A] [answered]
Do queues work in NativePHP?

Can you show an example?

**Comment by EPuigdemont** [comment]
I didn't find any documentation regarding the queues or how to start them, I got around this by adding an Artisan command schedule to my Console/Kernel.php file to execute the queue:work command every ten seconds without overlapping and closing on exit:

`$schedule->command('queue:work --stop-when-empty')->everyTenSeconds()->withoutOverlapping();`

This will call the command and have the queue:work command running until it exits, ensuring no double instances.
Adjust the timer as you like.

**Reply by simonhamp** [reply]
Docs for [Queues](https://nativephp.com/docs/1/digging-deeper/queues) was only added recently, but queueing has been part of NativePHP for a while.

**Comment by simonhamp** [comment, marked as answer]
We start [a single queue worker](https://github.com/NativePHP/electron-plugin/blob/main/src/server/index.ts#L31-L35) when your app loads up.

You shouldn't need to do anything special to queue jobs and to have them execute as intended.
